src/main/daml/Daml/Finance/Claims/Util/Builders.daml

```diff
@@ -192,3 +193,20 @@ createOptionsClaim dateToTime ownerReceives strike spot cashInstrumentCid expiry
     claim = if ownerReceives then claimAmount else give claimAmount
   in
     prepareAndTagClaims dateToTime [claim] "Option payment"
+
+-- | Create the claim for a physically settled European option.
+createEuropeanPhysicalClaim : (Date -> Time) -> Bool -> Decimal -> Deliverable -> Deliverable ->
+  Date -> Bool -> TaggedClaim
+createEuropeanPhysicalClaim dateToTime ownerReceives strike underlying ccy
+  expiryDate call =
+  let
+    exerciseClaim = if call then
+                      and (one underlying) $ scale (Const strike) $ give (one ccy)
+                    else
+                      and (give (one underlying)) $ scale (Const strike) $ (one ccy)
+    exercised = ("EXERCISE", exerciseClaim)
+    notExercised = ("EXPIRE", zero)
+    claimAmount = european expiryDate $ exercised `or` notExercised
```

Comment:
How does the final tree look like? I was under the assumption that `european` already includes the `or` node

Reply:
The TaggedClaim currently looks like this:
`TaggedClaim {claim = When (TimeGte 2019-05-15T00:00:00Z) (Or (EXERCISE: And (One InstrumentKey {depository = 'Custodian', issuer = 'Issuer', id = AAPL, version = "0"}) (Scale (Const 40.0) (Give (One InstrumentKey {depository = 'Custodian', issuer = 'Issuer', id = USD, version = "0"}))) []) (EXPIRE: Zero) [EXPIRE: Zero]), tag = "Option payment"}
`

Reply:
yup, `EXPIRE: Zero`  is added twice --> no need to add the `notExercised` part if you are using the `european` builder. You also don't have to pass any label, as those are added by the function itself.

Reply:
I based this on the example in `src/test/daml/Daml/Finance/Instrument/Generic/Test/EuropeanOption.daml`, which also contains the `or` as an argument to the `european` function.

Reply:
Thanks! I had a quick look and that test should indeed also be refactored to avoid the unnecessary branches

Reply:
> yup, `EXPIRE: Zero` is added twice --> no need to add the `notExercised` part if you are using the `european` builder. You also don't have to pass any label, as those are added by the function itself.

I can fix that in the Generic test as well.

Reply:
thank you for that

Reply:
I fixed this in both places now.